pyqtgraph/graphicsItems/HistogramLUTItem.py

```diff
@@ -25,177 +26,268 @@
+        'rgba'
+            One set of levels is drawn for each channel.
+    gradientPosition : str, optional
+        Position of the gradient editor relative to the histogram, either 'right' or
+        'left'.
+
+    Attributes
+    ----------
+    sigLookupTableChanged : signal
+        Emits the HistogramLUTItem itself when the gradient changes
+    sigLevelsChanged : signal
+        Emits the HistogramLUTItem itself while the movable region is changing
+    sigLevelChangeFinished : signal
+        Emits the HistogramLUTItem itself when the movable region is finished changing
+
+    See Also
+    --------
+    :class:`~pyqtgraph.ImageItem`
+        HistogramLUTItem is most useful when paired with an ImageItem.
+
+    :class:`~pyqtgraph.ImageView`
+        Widget containing a paired ImageItem and HistogramLUTItem.
+
+    :class:`~pyqtgraph.HistogramLUTWidget`
+        QWidget containing a HistogramLUTItem for widget-based layouts.
     """
-    
+
     sigLookupTableChanged = QtCore.Signal(object)
     sigLevelsChanged = QtCore.Signal(object)
     sigLevelChangeFinished = QtCore.Signal(object)
-    
-    def __init__(self, image=None, fillHistogram=True, rgbHistogram=False, levelMode='mono', gradientPosition='right'):
+
+    def __init__(self, image=None, fillHistogram=True, levelMode='mono',
+                 gradientPosition='right', orientation='vertical'):
         GraphicsWidget.__init__(self)
         self.lut = None
         self.imageItem = lambda: None  # fake a dead weakref
         self.levelMode = levelMode
-        self.rgbHistogram = rgbHistogram
+        self.orientation = orientation
         self.gradientPosition = gradientPosition
-        
+
+        if orientation == 'vertical' and gradientPosition not in {'right', 'left'}:
+            self.gradientPosition = 'right'
+        elif orientation == 'horizontal' and gradientPosition not in {'top', 'bottom'}:
+            self.gradientPosition = 'bottom'
+
         self.layout = QtGui.QGraphicsGridLayout()
         self.setLayout(self.layout)
-        self.layout.setContentsMargins(1,1,1,1)
+        self.layout.setContentsMargins(1, 1, 1, 1)
         self.layout.setSpacing(0)
+
         self.vb = ViewBox(parent=self)
-        self.vb.setMaximumWidth(152)
-        self.vb.setMinimumWidth(45)
-        self.vb.setMouseEnabled(x=False, y=True)
-        self.gradient = GradientEditorItem()
-        self.gradient.setOrientation(gradientPosition)
+        if self.orientation == 'vertical':
+            self.vb.setMaximumWidth(152)
+            self.vb.setMinimumWidth(45)
+            self.vb.setMouseEnabled(x=False, y=True)
+        else:
+            self.vb.setMaximumHeight(152)
+            self.vb.setMinimumHeight(45)
+            self.vb.setMouseEnabled(x=True, y=False)
+
+        self.gradient = GradientEditorItem(orientation=self.gradientPosition)
         self.gradient.loadPreset('grey')
+
+        ori = 'horizontal' if self.orientation == 'vertical' else 'vertical'
```

Comment:
is this right?  shouldn't it be

```python
ori = 'horizontal if self.orientation == 'horizontal' else 'vertical'
```

Reply:
Yeah I considered adding a comment here because it's confusing. For `LinearRegionItem`, orientation refers to the orientation of the *lines*.

Reply:
Oh right, may I suggest renaming the variable to `regionOrientation` or something along those lines.